lib/ability.rb

```diff
@@ -19,6 +19,7 @@ def initialize(user, resource, controller = nil)
         can :manage, UserPriceGroupMember if resource.admin_editable?
         can :manage, AccountPriceGroupMember
       else
+        can([:edit, :update], User) { |target_user| target_user.external? }
```

Comment:
You're on the right track, but I think you may need to flip this into a `cannot`, and move it after the `can :manage, :all` below. But I'd start with adding tests for these ability in `spec/lib/ability_spec.rb`